File: server/router/lgjx/testInvoiceApply.go
package lgjx

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type TestInvoiceApplyRouter struct {
}

func (s *TestInvoiceApplyRouter) InitInvoiceApplyRouter(Router *gin.RouterGroup) {
	testInvoiceApplyRouter := Router.Group("testInvoiceApply").Use(middleware.OperationRecord())
	testInvoiceApplyRouterWithoutRecord := Router.Group("testInvoiceApply")
	var testInvoiceApplyApi = v1.ApiGroupApp.LgjxTestApiGroup.TestInvoiceApplyApi
	{
		testInvoiceApplyRouter.POST("createInvoiceApply", testInvoiceApplyApi.CreateInvoiceApply)             // 新建InvoiceApply
		testInvoiceApplyRouter.DELETE("deleteInvoiceApply", testInvoiceApplyApi.DeleteInvoiceApply)           // 删除InvoiceApply
		testInvoiceApplyRouter.DELETE("deleteInvoiceApplyByIds", testInvoiceApplyApi.DeleteInvoiceApplyByIds) // 批量删除InvoiceApply
		testInvoiceApplyRouter.PUT("updateInvoiceApply", testInvoiceApplyApi.UpdateInvoiceApply)              // 更新InvoiceApply
	}
	{
		testInvoiceApplyRouterWithoutRecord.GET("findInvoiceApply", testInvoiceApplyApi.FindInvoiceApply)       // 根据ID获取InvoiceApply
		testInvoiceApplyRouterWithoutRecord.GET("getInvoiceApplyList", testInvoiceApplyApi.GetInvoiceApplyList) // 获取InvoiceApply列表
	}
}

File: web/src/api/testInvoiceApply.js
import service from '@/utils/request'

export const createInvoiceApply = (data) => {
  return service({
    url: '/testInvoiceApply/createInvoiceApply',
    method: 'post',
    data
  })
}

export const deleteInvoiceApply = (data) => {
  return service({
    url: '/testInvoiceApply/deleteInvoiceApply',
    method: 'delete',
    data
  })
}

export const deleteInvoiceApplyByIds = (data) => {
  return service({
    url: '/testInvoiceApply/deleteInvoiceApplyByIds',
    method: 'delete',
    data
  })
}

export const updateInvoiceApply = (data) => {
  return service({
    url: '/testInvoiceApply/updateInvoiceApply',
    method: 'put',
    data
  })
}

export const findInvoiceApply = (params) => {
  return service({
    url: '/testInvoiceApply/findInvoiceApply',
    method: 'get',
    params
  })
}

export const getInvoiceApplyList = (params) => {
  return service({
    url: '/testInvoiceApply/getInvoiceApplyList',
    method: 'get',
    params
  })
}

export const approveInvoiceApply = (data) => {
  return service({
    url: '/testInvoiceApply/approveInvoiceApply',
    method: 'put',
    data
  })
}

export const rejectInvoiceApply = (data) => {
  return service({
    url: '/testInvoiceApply/rejectInvoiceApply',
    method: 'put',
    data
  })
}